includes/libraries/tracks/class-tracks-client.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+/**
+ * Send Tracks events on behalf of a user.
+ *
+ * @class   Tracks_Client
+ * @package WooCommerce/Classes
+ *
+ *
+ * Example Usage:
+```php
+	require( dirname(__FILE__).'path/to/tracks/class-tracks-client.php' );
+
+	$result = Tracks_Client::record_event( array(
+		'_en'        => $event_name,       // required
+		'_ui'        => $user_id,          // required unless _ul is provided
+		'_ul'        => $user_login,       // required unless _ui is provided
+
+		// Optional, but recommended
+		'_ts'        => $ts_in_ms,         // Default: now
+		'_via_ip'    => $client_ip,        // we use it for geo, etc.
+
+		// Possibly useful to set some context for the event
+		'_via_ua'    => $client_user_agent,
+		'_via_url'   => $client_url,
+		'_via_ref'   => $client_referrer,
+	) );
+
+	if ( is_wp_error( $result ) ) {
+		// Handle the error in your app
+	}
+```
+ */
+
+require_once dirname( __FILE__ ) . '/class-tracks-event.php';
+
+/**
+ * Class Tracks_Client.
+ */
+class Tracks_Client {
+	const PIXEL           = 'https://pixel.wp.com/t.gif';
+	const BROWSER_TYPE    = 'php-agent';
+	const USER_AGENT_SLUG = 'tracks-client';
+
+	/**
+	 * Record a Tracks event
+	 *
+	 * @param  mixed $event Event object to send to Tracks. An array will be cast to object. Required.
+	 *                      Properties are included directly in the pixel query string after light validation.
+	 * @return mixed         True on success, WP_Error on failure
+	 */
+	public static function record_event( $event ) {
+		if ( ! $event instanceof Tracks_Event ) {
+			$event = new Tracks_Event( $event );
+		}
+		if ( is_wp_error( $event ) ) {
+			return $event;
+		}
+
+		$pixel = $event->build_pixel_url( $event );
+
+		if ( ! $pixel ) {
+			return new WP_Error( 'invalid_pixel', 'cannot generate tracks pixel for given input', 400 );
+		}
+
+		return self::record_pixel( $pixel );
+	}
+
+	/**
+	 * Synchronously request the pixel.
+	 *
+	 * @param string $pixel pixel url and query string.
+	 * @return mixed An error or true if successful.
+	 */
+	public static function record_pixel( $pixel ) {
+		// Add the Request Timestamp and URL terminator just before the HTTP request.
+		$pixel .= '&_rt=' . self::build_timestamp() . '&_=_';
+
+		$response = wp_remote_get(
+			$pixel,
+			array(
+				'blocking'    => true, // The default, but being explicit here.
+				'timeout'     => 1,
+				'redirection' => 2,
+				'httpversion' => '1.1',
+			)
+		);
+
+		if ( is_wp_error( $response ) ) {
+			return $response;
+		}
+
+		$code = isset( $response['response']['code'] ) ? $response['response']['code'] : 0;
+
+		if ( 200 !== $code ) {
+			return new WP_Error( 'request_failed', 'Tracks pixel request failed', $code );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Build an event and return its tracking URL
+	 *
+	 * @deprecated          Call the `build_pixel_url` method on a Tracks_Event object instead.
```

Comment:
Why this already comes with deprecated methods?
As I said before, this library seems at least know built for WooCommerce...